ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -831,9 +832,80 @@ prop_TxSeq_lookupByTicketNo_sound smalls small =
     txseq =
         foldl' (TxSeq.:>) TxSeq.Empty $ map mkTicket haystack
 
-    mkTicket x = TxTicket x (mkTicketNo x)
+    mkTicket x = TxTicket x (mkTicketNo x) 0
     mkTicketNo = TicketNo . toEnum
 
+-- | Test that the 'fst' of the result of 'splitAfterTxSize' only contains
+-- 'TxTicket's whose summed up transaction sizes are less than or equal to
+-- that of the 'TxSizeInBytes' which the 'TxSeq' was split on.
+prop_TxSeq_splitAfterTxSize :: TxSizeSplitTestSetup -> Property
+prop_TxSeq_splitAfterTxSize tss =
+      property $ txSizeSum (txTickets before) <= tssTxSizeToSplitOn
+  where
+    TxSizeSplitTestSetup { tssTxSizeToSplitOn } = tss
+
+    (before, _after) = splitAfterTxSize txseq tssTxSizeToSplitOn
+
+    txseq :: TxSeq Int
+    txseq = txSizeSplitTestSetupToTxSeq tss
+
+    txSizeSum :: [TxTicket tx] -> TxSizeInBytes
+    txSizeSum = sum . map txTicketTxSizeInBytes
+
+
+-- | Test that the results of 'splitAfterTxSizeSpec', a specification of
+-- 'splitAfterTxSize', match those of the real 'splitAfterTxSize'
+-- implementation.
+prop_TxSeq_splitAfterTxSizeSpec :: TxSizeSplitTestSetup -> Property
+prop_TxSeq_splitAfterTxSizeSpec tss =
+    splitAndCheck tssTxSizeToSplitOn
+  where
+    TxSizeSplitTestSetup { tssTxSizeToSplitOn } = tss
+
+    txseq :: TxSeq Int
+    txseq = txSizeSplitTestSetupToTxSeq tss
+
+    splitAndCheck :: TxSizeInBytes -> Property
+    splitAndCheck txsz =
+      let (implBefore, implAfter) = splitAfterTxSize txseq txsz
+          (specBefore, specAfter) = splitAfterTxSizeSpec txseq txsz
+      in      txTickets implBefore === txTickets specBefore
+         .&&. txTickets implAfter  === txTickets specAfter
+
+{-------------------------------------------------------------------------------
+  TxSizeSplitTestSetup
+-------------------------------------------------------------------------------}
+
+data TxSizeSplitTestSetup = TxSizeSplitTestSetup
+  { tssTxSizes         :: ![TxSizeInBytes]
+  , tssTxSizeToSplitOn :: !TxSizeInBytes
+  } deriving Show
+
+instance Arbitrary TxSizeSplitTestSetup where
+  arbitrary = do
+    let txSizeMaxBound = 10000000 -- 10MB transaction max bound
+    txSizes <- listOf $ choose (1, txSizeMaxBound)
+    let totalTxSize = sum txSizes
+    txSizeToSplitOn <- choose (0, totalTxSize)
```

Comment:
Just to make sure we're also testing the boundaries:
```haskell
frequency 
  [ (1, pure 0)
  , (8, choose (0, totalTxsSize))
  , (1, pure totalTxsSize)
  ]
```
(I would also rename `totalTxSize` to `totalTxsSize`)
There is some overlap now, but `choose (1, totalTxsSize - 1)` could underflow if `totalTxsSize == 0`, but you can change the generator if you prefer.

We should also test with a `txsSizeToSplitOn > totalTxsSize`:
```haskell
  , (1, choose (totalTxsSize + 1, totalTxsSize + 1000))
```

Reply:
For the generation of `txSizes`, I've instead utilized `listOf1` to avoid the possibility of `sum txSizes == 0`.

Reply:
No, `listOf` is fine, we want to correctly handle that case too.